### PR TITLE
CHM345_CHM344-fix map can't work issue, add treaty for chm

### PR DESCRIPTION
**Pull request by liyuanmontreal**
### General description
The maps on CHM home page and country profile page  are not working, 
the requirements is to fix it , let the map is working as intended like on https://absch.cbddev.xyz/ 

### Designs
the reason of this issue is lack chm related map files in project. 
so add realm for chm
add variable, condition judge and template, js ,json file for chm-map 
it also contains the change for CHM344( add treaty for chm)

### Testing instructions
test homepage- the whole map and records table  on right side display correctly.
click one country , the map and records table display correctly
go to country page, the info display correctly

###To-do
these two files use same content as ABS, need to change to chm content.
homepin-popup-chm.json file 
pin-popup-chm.json file 
 

https://github.com/scbd/absch.cbd.int/assets/65098066/a0d7cdc3-e764-4d42-9947-fcf9dfb75eea

### CODE REVIEW

**Comment by liyuanmontreal** [comment]
update countries list page
image

File: app/views/countries/country-map.js [hunk 1]
```diff
@@ -96,7 +97,10 @@ import countryMapTranslation from '~/app-text/views/countries/country-map.json';
           }
         }
         else if(realm.is('ABS')){
-          $scope.isABS          = true;
+          $scope.isABS          = true;          
+        }
+        else if(realm.is('CHM')){
+          $scope.isCHM          = true;       
         }
```
Comment (blaisefonseca) [inline app/views/countries/country-map.js:104]:
```suggestion
else {
          $scope.isABS          = realm.is('ABS');          
          $scope.isCHM          = realm.is('CHM');       
        }
```

Comment (liyuanmontreal) [reply app/views/countries/country-map.js:109]: done

**Comment by liyuanmontreal** [comment]
image

File: app/views/countries/country-map.html [hunk 2]
```diff
@@ -197,15 +197,18 @@ <h2 class="modal-title">{{::('countryMapTranslation.disclaimer'|$translate)}}</h
...
                         <i class="flag-icon flag-icon-{{currentCountry.code.toLowerCase()}}"></i>
-                        <span ng-bind="currentCountry.name|lstring"></span>
+                        <span ng-bind="currentCountry.name|lstring"></span>                        
                     </a>
                 </div>
                     <div ng-if="isBCH && currentCountry">
                         <homepin-popup-bch>
+                    </div>                   
+                    <div ng-if="isCHM && currentCountry">                          
+                        <home-pin-popup-chm ng-vue="vueComponent" :country="currentCountry" is-chm="isCHM"></home-pin-popup-chm>
```
Comment (blaisefonseca) [inline app/views/countries/country-map.html:211]:
```suggestion
                        <home-pin-popup-chm ng-vue="vueComponent" :country="currentCountry"></home-pin-popup-chm>                         
```

File: app/views/countries/directives/home-pin-popup-chm.vue [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+</template>
+
+<script setup>
+    import messages from '~/app-text/views/countries/directives/home-pin-popup-chm.json';
+    import { useRoute } from "@scbd/angular-vue/src/index.js";  
+    import { useI18n } from 'vue-i18n';    
+    import { lstring } from '../../../components/kb/filters';
+ 
+  
+    //const route = useRoute().value;
```
Comment (blaisefonseca) [inline app/views/countries/directives/home-pin-popup-chm.vue:49]:
```suggestion
```

File: app/views/countries/directives/home-pin-popup-chm.vue [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    import { useI18n } from 'vue-i18n';    
+    import { lstring } from '../../../components/kb/filters';
+ 
+  
+    //const route = useRoute().value; 
+    const {value:route} = useRoute();    
+    const { t,locale } = useI18n({ messages }); 
+
+    const props = defineProps({
+        country: {type: Object, require:true, default:{}}
```
Comment (blaisefonseca) [inline app/views/countries/directives/home-pin-popup-chm.vue:54]: when props are required, should not have default value
```suggestion
        country: {type: Object, require:true}     
```

File: app/views/countries/directives/home-pin-popup-chm.vue [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  
+    //const route = useRoute().value; 
+    const {value:route} = useRoute();    
+    const { t,locale } = useI18n({ messages }); 
+
+    const props = defineProps({
+        country: {type: Object, require:true, default:{}}     
+    })
+
+    const ToggleCountyList = (code, currentCountryCode) => {
```
Comment (blaisefonseca) [inline app/views/countries/directives/home-pin-popup-chm.vue:57]:
```suggestion
    const toggleCountyList = (code, currentCountryCode) => {
```

File: app/views/countries/directives/home-pin-popup-chm.vue [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+        <div>
+            <table class="table table-map table-condensed" style="display: block;overflow-x: auto;white-space: nowrap;" >           
+           
+                <tr v-for="schema in country.schemas" style="border:none;">            
+                    <td>
+                        <span class="rounded badge ms-2 "  :class="{'bg-gray-300' : schema.count==0, 'bg-chm':schema.count>0 }" >{{schema.count}}</span>
+                    </td>
+                    <td >                  
+                        <a rel="noopener" v-if="country.code"  @click="ToggleCountyList(schema.shortCode,country.code)" class="text-black cursor-pointer" style="color:#fff" >
```
Comment (blaisefonseca) [inline app/views/countries/directives/home-pin-popup-chm.vue:29]:
```suggestion
                        <a rel="noopener" v-if="country.code"  @click="toggleCountyList(schema.shortCode,country.code)" class="text-black cursor-pointer" style="color:#fff" >
```

File: app/views/countries/directives/home-pin-popup-chm.vue [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        if ((route?.params?.code) && $("#div" + code).length>0) {
+            $('html, body').animate({
+                scrollTop: $("#div"+code).offset().top
+            }, 500);
+            if ($("#div" + code).find('i').hasClass("bi-chevron-down")){
+                $("#div" + code).trigger('click');
+            }
+        }
+        else{
+            route?.params?.path(`countries/${currentCountryCode}/${code}`);
```
Comment (blaisefonseca) [inline app/views/countries/directives/home-pin-popup-chm.vue:67]: you need to use $router.push

**Review by blaisefonseca — APPROVED** [review]

**Pull request merged**